File: SWE/LeetCode/0001_TwoSum/0001_TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Time: O(n), Space: O(n)
        map<int,int> m;
        vector<int> ans(2);
        for (int i = 0; i < nums.size(); i++) {
            int sub = target - nums[i];
            if (m.find(sub) != m.end()) {
                ans[0] = i;
                ans[1] = m[sub];
                return ans;
            }
            m[nums[i]] = i;
        }
        return ans;
    }
};

File: SWE/LeetCode/0001_TwoSum/0001_TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Time: O(n), Space: O(n)
        d = {}
        for i in range(len(nums)):
            sub = target - nums[i]
            if sub in d:
                return [i, d[sub]]
            d[nums[i]] = i
        return [0, 0]
